File: JazSharpFirstImpressions_Source/LM.JazSharp.Examples/LM.JazSharp.Examples.App/Adder.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LM.JazSharp.Examples.App
{
    class Adder
    {
        private readonly OperationHistory _operationHistory;
        public Adder(OperationHistory existingHistory = null)
        {
            _operationHistory = existingHistory ?? new OperationHistory();
        }

        public decimal Add(decimal n1, decimal n2)
        {
            _operationHistory.Append($"{n1} + {n2}");

            return n1 + n2;
        }
    }
}

File: JazSharpFirstImpressions_Source/lm-jasmine-examples/Adder.js
class Adder {
    operationHistory;

    constructor(existingHistory) {
        this.operationHistory = existingHistory || new OperationHistory();
    }

    add(n1, n2) {
        this.operationHistory.append(`${n1} + ${n2}`);
        return n1 + n2;
    } 
}

class OperationHistory {
    operations = new Array();
    
    append(operation) {
        this.operations.push(operation);
    }

    count() {
        return this.operations.length;
    }
}